docs/rackhd/tftp_dhcp_server.rst

```diff
@@ -0,0 +1,313 @@
+all together.
+
+There are various Linux DHCP Server versions out there, RackHD has been primarily
+validated against `ISC DHCP Server`_. As long as the DHCP server supports the required
+DHCP configuration options then those versions should be compatible as well.
+
+Default ISC DHCP + on-dhcp-proxy Configuration
+""""""""""""""""""""""""""""""""""""""""""""""
+
+The advantage of using the on-dhcp-proxy service is to avoid complicated DHCP server setup,
+most logics are handled in on-dhcp-proxy, it's convinient and flexiable,
+but a typical simple `dhcpd.conf`_ of `ISC DHCP Server`_ for forwarding DHCP request to RackHD's
+on-dhcp-proxy service is needed, it would look like the following:
+
+.. literalinclude:: samples/dhcpd.conf
+
+.. _subnet: https://en.wikipedia.org/wiki/Subnetwork
+.. _range: https://en.wikipedia.org/wiki/Dynamic_Host_Configuration_Protocol#Overview
+.. _netmask: https://en.wikipedia.org/wiki/Subnetwork#Network_addressing_and_routing
+
+Subsituting the `subnet`_, `range`_ and `netmask`_ to match your desired networking configuration.
+
+To enforce lease assignment based on MAC and not UID we opt-in to ignore the UID in the request
+by setting **ignore-client-uids true**.
+
+ISC DHCP Only Configuration
+"""""""""""""""""""""""""""
+
+When bypassing on-dhcp-proxy step, the DHCP server can also define static host definitions.
+
+Using `ISC DHCP Server`_, the `dhcpd.conf`_ for invoking RackHD lookup directly
+and handling of static host definitions would look like the following:
+
+.. literalinclude:: samples/dhcpd.conf.noproxy
+
+In the global subnet definition we define a PXE chainloading setup to handle specific
+client requests.
+
+.. code-block:: shell
+
+    if ((exists user-class) and (option user-class = "MonoRail")) {
+        ...
+    } else {
+        ...
+    }
+
+If the request is made from a BIOS/UEFI PXE client, the DHCP server will hand out the
+iPXE bootloader image that corresponds to the system's architecture type.
+
+.. code-block:: shell
+
+    if ((exists user-class) and (option user-class = "MonoRail")) {
+        filename "http://172.31.128.1:9080/api/common/profiles";
+    } else {
+        if option arch-type = 00:09 {
+          filename "monorail-efi64-snponly.efi";
+        } elsif option arch-type = 00:07 {
+          filename "monorail-efi64-snponly.efi";
+        } elsif option arch-type = 00:06 {
+          filename "monorail-efi32-snponly.efi";
+        } else {
+          filename "monorail.ipxe";
+        }
+    }
+
+If the request is made from the RackHD iPXE client, the DHCP server will chainload
+another boot configuration pointed at RackHD's profiles API.
+
+Third-Party DHCP Service Configuration
+""""""""""""""""""""""""""""""""""""""
+
+.. _Default iPXE Config: https://github.com/RackHD/on-imagebuilder/blob/master/roles/ipxe/files/default.ipxe
+
+In theory, Other DHCP services are also supported by RackHD, but it's not validated yet, possible solutions are as below:
```

Comment:
"In theory, Other DHCP services are also supported by RackHD, but it's not validated yet, possible solutions are as below:" --> "Other DHCP services may be used with possible solution configurations below:"